projects/com.oracle.truffle.llvm.parser/src/com/oracle/truffle/llvm/parser/model/enums/BinaryOperator.java

```diff
@@ -61,12 +61,23 @@ public static BinaryOperator decode(int opcode, boolean isFloatingPoint) {
     }
 
     private final int fpmap;
+    private final String name;
 
-    BinaryOperator(int fpmap) {
+    BinaryOperator(int fpmap, String name) {
         this.fpmap = fpmap;
+        this.name = name;
     }
 
     private BinaryOperator fp() {
         return fpmap < 0 ? null : values()[fpmap];
     }
+
+    public boolean isFloatingPoint() {
+        return this.ordinal() > INT_XOR.ordinal();
```

Comment:
This is a public API function that is not used in Sulong. Therefore future refactorings might remove this function.
Please add some JavaDoc/comment why this function exists and what its purpose is.